apps/web/src/components/Basenames/UsernameProfileTransferOwnershipModal/context.tsx

```diff
@@ -0,0 +1,387 @@
+    OwnershipSteps.Search,
+  );
+
+  // Send calls - Experimental
+  const { initiateBatchCalls, batchCallsEnabled, batchCallsStatus } = useSendCallsWithLogs({
+    chain: basenameChain,
+    eventName: 'basename_send_calls_transfer_ownership',
+  });
+
+  // TODO: Validate that it's not a contract recipient
+  const isValidRecipientAddress = isAddress(recipientAddress);
+  const tokenId = getTokenIdFromBasename(profileUsername);
+
+  // The 4 transactions we got to track
+  const {
+    initiateTransaction: initiateSafeTransferFrom,
+    transactionStatus: safeTransferFromStatus,
+  } = useWriteContractWithReceipt({
+    chain: basenameChain,
+    eventName: 'basename_safe_transfer_from',
+  });
+
+  const { initiateTransaction: initiateReclaim, transactionStatus: reclaimStatus } =
+    useWriteContractWithReceipt({
+      chain: basenameChain,
+      eventName: 'basename_reclaim',
+    });
+
+  const { initiateTransaction: initiateSetAddr, transactionStatus: setAddrStatus } =
+    useWriteContractWithReceipt({
+      chain: basenameChain,
+      eventName: 'basename_set_addr',
+    });
+
+  const { initiateTransaction: initiateSetName, transactionStatus: setNameStatus } =
+    useWriteContractWithReceipt({
+      chain: basenameChain,
+      eventName: 'basename_set_name',
+    });
+
+  const updateViaBatchCalls = useCallback(async () => {
+    if (!isValidRecipientAddress) return;
+    if (!address) return;
+    if (!basenameChain) return;
+    if (!batchCallsEnabled) return;
+
+    await initiateBatchCalls({
+      calls: [
+        {
+          to: USERNAME_L2_RESOLVER_ADDRESSES[basenameChain.id],
+          data: encodeFunctionData({
+            abi: L2Resolver,
+            args: [namehash(profileUsername), recipientAddress],
+            functionName: 'setAddr',
+          }),
+        },
+        {
+          to: USERNAME_BASE_REGISTRAR_ADDRESSES[basenameChain.id],
+          data: encodeFunctionData({
+            abi: BaseRegistrarAbi,
+            args: [tokenId, recipientAddress],
+            functionName: 'reclaim',
+          }),
+        },
+        {
+          to: USERNAME_BASE_REGISTRAR_ADDRESSES[basenameChain.id],
+          data: encodeFunctionData({
+            abi: BaseRegistrarAbi,
+            args: [address, recipientAddress, tokenId],
+            functionName: 'safeTransferFrom',
+          }),
+        },
+        {
+          to: USERNAME_REVERSE_REGISTRAR_ADDRESSES[basenameChain.id],
+          data: encodeFunctionData({
+            abi: ReverseRegistrarAbi,
+            args: [''],
+            functionName: 'setName',
+          }),
+        },
+      ],
+      account: address,
+      chain: basenameChain,
+    });
+  }, [
+    address,
+    basenameChain,
+    batchCallsEnabled,
+    initiateBatchCalls,
+    isValidRecipientAddress,
+    profileUsername,
+    recipientAddress,
+    tokenId,
+  ]);
+
+  // The 4 Function with safety checks
+  const updateSetAddr = useCallback(async () => {
```

Comment:
I think you might be able to drop the async/await on these and get the same result

Reply:
WDYM?

Reply:
Like, this could be written as 

```
() => {
    if (!setAddrContract) return Promise.reject('Invalid setAddrContract');
    return initiateSetAddr(setAddrContract);
}
 ```
 
 However I don't mind having things explicitly marked as async, it helps my dumb brain.

Reply:
ight will keep as is.